### PR TITLE
getitem only returns top variable

### LINKED ISSUE #786: Model.__getitem__ should return highest-level varkey
list of varkeys all with the same name should be in `.byname`. Also there could be a fuzzy search, `.search("W wing")` which would get all of `W_wing`, `W with model=['wing']`, `W descr="wing weight"` and return a list sorted by relevance (or just the most relevant)

### PR DESCRIPTION
**Pull request by bqpd**
closes #786 

does not work for vectorvariables.

### CODE REVIEW

File: gpkit/constraints/set.py [hunk 1]
```diff
@@ -5,12 +5,6 @@
 from ..repr_conventions import _str, _repr, _repr_latex_
 
 
-def _var_sort_key(var):
-    "return thing to compare for Variable sorting"
-    k = var.key
-    return (k.str_without(["units", "idx"]), k.idx)
-
-
```
Comment (whoburg) [inline gpkit/constraints/set.py]: this fixed #724; I want to retain some functionality (not necessarily `__getitem__` for getting a correctly-sorted VectorVariable from a model.

File: gpkit/constraints/set.py [hunk 2]
```diff
@@ -44,8 +38,11 @@ def __getitem__(self, key):
             if len(variables) == 1:
                 return variables[0]
             else:
-                variables.sort(key=_var_sort_key)
-                return variables
+                sortfn = lambda v: len(v.key.models) if v.key.models else 0
```
Comment (whoburg) [inline gpkit/constraints/set.py]: http://stackoverflow.com/questions/25010167/e731-do-not-assign-a-lambda-expression-use-a-def

File: gpkit/constraints/set.py [hunk 3]
```diff
@@ -44,8 +38,11 @@ def __getitem__(self, key):
...
+                variables.sort(key=sortfn)
+                variable = variables[0]
+                # note: doesn't work for vector variables
```
Comment (whoburg) [inline gpkit/constraints/set.py]: what is the plan for VectorVariables?

File: gpkit/constraints/set.py [hunk 4]
```diff
@@ -44,8 +38,11 @@ def __getitem__(self, key):
...
+                return variable
```
Comment (whoburg) [inline gpkit/constraints/set.py]: shouldn't we also retain the old `__getitem__` functionality, that returned multiple search hits, in a new method?

Comment (bqpd) [inline gpkit/constraints/set.py]: Yes, can't do it today.

File: gpkit/constraints/set.py (hunk 3 above)
Comment (bqpd) [inline gpkit/constraints/set.py]: Probably they should be stored as arrays, not by their individual varkeys; but that's a issue which predates this and should have a new issue.

File: gpkit/constraints/set.py (hunk 2 above)
Comment (bqpd) [inline gpkit/constraints/set.py]: Indeed.

File: gpkit/constraints/set.py (hunk 1 above)
Comment (bqpd) [inline gpkit/constraints/set.py]: Great, probably can't do it until this evening.

File: gpkit/constraints/set.py (hunk 2 above)
Comment (bqpd) [inline gpkit/constraints/set.py]: Done.

File: gpkit/constraints/set.py (hunk 4 above)
Comment (bqpd) [inline gpkit/constraints/set.py]: Done.

File: gpkit/constraints/set.py (hunk 1 above)
Comment (bqpd) [inline gpkit/constraints/set.py]: Done in `variables_byname`

File: gpkit/constraints/set.py (hunk 3 above)
Comment (bqpd) [inline gpkit/constraints/set.py]: issue #799

File: gpkit/constraints/set.py [hunk 5]
```diff
@@ -5,6 +5,16 @@
 from ..repr_conventions import _str, _repr, _repr_latex_
 
 
+def _sort_by_num_models(var):
+    "return integer for Variable sorting"
+    return len(v.key.models) if v.key.models else 0
```
Comment (whoburg) [inline gpkit/constraints/set.py]: possible premature optimization, but would it be faster to not lookup `v.key.models` twice?

```python
def _sort_by_num_models(var):
    "return integer for Variable sorting"
    mods = v.key.models
    return len(mods) if mods else 0
```

File: .pylintrc [hunk 6]
```diff
@@ -61,7 +61,7 @@ confidence=
 # --disable=W"
 #disable=import-star-module-level,old-octal-literal,oct-method,print-statement,unpacking-in-except,parameter-unpacking,backtick,old-raise-syntax,old-ne-operator,long-suffix,dict-view-method,dict-iter-method,metaclass-assignment,next-method-called,raising-string,indexing-exception,raw_input-builtin,long-builtin,file-builtin,execfile-builtin,coerce-builtin,cmp-builtin,buffer-builtin,basestring-builtin,apply-builtin,filter-builtin-not-iterating,using-cmp-argument,useless-suppression,range-builtin-not-iterating,suppressed-message,no-absolute-import,old-division,cmp-method,reload-builtin,zip-builtin-not-iterating,intern-builtin,unichr-builtin,reduce-builtin,standarderror-builtin,unicode-builtin,xrange-builtin,coerce-method,delslice-method,getslice-method,setslice-method,input-builtin,round-builtin,hex-method,nonzero-method,map-builtin-not-iterating
 
-disable=superfluous-parens,locally-disabled,misplaced-comparison-constant,too-few-public-methods,no-self-use
+disable=superfluous-parens,locally-disabled,misplaced-comparison-constant,too-few-public-methods,no-self-use,invalid-sequence-index
```
Comment (whoburg) [inline .pylintrc]: hmm, but a model is a sequence, because it acts like a list, right? Where does this occur? Shouldn't we just disable locally?

**Comment by whoburg** [comment]
coming along; 2nd round review complete.

File: gpkit/constraints/set.py (hunk 5 above)
Comment (bqpd) [inline gpkit/constraints/set.py]: Heh, sure thing.

File: .pylintrc (hunk 6 above)
Comment (bqpd) [inline .pylintrc]: You can use strings to get variables out of models, and I don't want a pylint error every single time that happens...

**Pull request merged**